compiler/src/yul/mappers/contracts.rs

```diff
@@ -55,24 +59,28 @@ pub fn contract_def(
             vec![]
         };
 
-        return Ok(yul::Object {
-            name: identifier! { Contract },
-            code: constructor,
-            objects: vec![yul::Object {
-                name: identifier! { runtime },
-                code: yul::Code {
-                    block: yul::Block {
-                        statements: statements! {
-                            [user_functions...]
-                            [runtime...]
-                        },
+        let runtime = yul::Object {
+            name: identifier! { runtime },
+            code: yul::Code {
+                block: yul::Block {
+                    statements: statements! {
+                        [user_functions...]
+                        [runtime...]
+                        // we must return, otherwise we'll enter into other objects
+                        (return(0, 0))
                     },
                 },
-                objects: vec![],
-                // We can't reach to data objects in the "contract" hierachy so in order to have
-                // the data objects available in both places we have to put them in both places.
-                data: data.clone(),
-            }],
+            },
+            objects: external_contracts,
```

Comment:
these contract objects are not optimized out by the yul compiler, so they should only be added if `create` or `create2` are used